apps/genpkey.c

```diff
@@ -237,13 +237,13 @@ int genpkey_main(int argc, char **argv)
         goto end;
     }
 
-    out = bio_open_owner(outfile, outformat, private);
-    if (out == NULL)
+    mem_out = BIO_new(BIO_s_mem());
```

Comment:
On success, best to also call `BIO_set_mem_eof_return(mem_out, 0)`, this memory BIO will NOT be used for alternating reads and appends (for which indications to retry could be of some use).

Reply:
Updated.